src/Shared/Program.cs

```diff
@@ -84,24 +84,24 @@ public static void Main(string[] args)
                 }
 
                 string? output = null;
-                
+
                 // TODO: Remove STA check now that the methods are async anyway.
                 // TODO: Test... this seems backwards/opposite
                 if (method.GetCustomAttribute(typeof(STAThreadAttribute), false) is object)
                 {
-                    Task task = new Task(() =>
+                    Task task = new Task(async () =>
                     {
                         // TODO: Change to use async/await.
-                        output = InvokeMethodUsingReflection(method,arguments).GetAwaiter().GetResult();
+                        output = await InvokeMethodUsingReflectionAsync(method, arguments);
                     });
                     task.Wait();
                 }
                 else
                 {
                     // TODO: Change to use async/await.
-                    output = InvokeMethodUsingReflection(method, arguments).GetAwaiter().GetResult();
+                    output = await InvokeMethodUsingReflectionAsync(method, arguments);
```

Comment:
I think entire if-condition can be replaced with:
```C#
output = await InvokeMethodUsingReflectionAsync(method, arguments);
```

The extra Tasks don't appear to be adding anything, and the STAAttribute is getting ignored and wont be used anyway. If all of the tests are passing we should b esafe to remove it.